pkg/uio/progress.go

```diff
@@ -19,15 +19,49 @@ type ProgressReader struct {
 	W        io.Writer
 
 	counter int
+	written bool
 }
 
 // Read implements io.Reader for ProgressReader.
 func (r *ProgressReader) Read(p []byte) (n int, err error) {
 	defer func() {
-		//log.Print("r.Counter %d, r.Interval %d, n
 		numSymbols := (r.counter%r.Interval + n) / r.Interval
 		r.W.Write([]byte(strings.Repeat(r.Symbol, numSymbols)))
 		r.counter += n
+		r.written = (r.written || numSymbols > 0)
+		if err == io.EOF && r.written {
+			r.W.Write([]byte("\n"))
+		}
 	}()
 	return r.R.Read(p)
 }
+
+// ProgressReadCloser implements io.ReadCloser and prints Symbol to W after every
+// Interval bytes passes through RC.
+type ProgressReadCloser struct {
+	RC io.ReadCloser
+
+	Symbol   string
+	Interval int
+	W        io.Writer
+
+	pr *ProgressReader
+}
+
+// Read implements io.Reader for ProgressReader.
+func (rc *ProgressReadCloser) Read(p []byte) (n int, err error) {
+	if rc.pr == nil {
```

Comment:
This will, I think, get the race detector very upset. I think this will need a different approach.

Reply:
Does the io.Reader guarantee thread-safety? Maybe people assume so if it wraps an os.File.

Reply:
The io.Reader doesn't guarantee thread-safety. This isn't any less thread-safe than before and the race detector isn't complaining.